### PR TITLE
Updates LightGBM from 2.X.X to 3.X.X

### PR DESCRIPTION
**Pull request by michaelgsharp**
Updates LightGBM from 2.X.X to 3.X.X and all the associated baseline test files.

Also updates the SQL nuget package for its security update.

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <add key="dotnet5-roslyn" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
     <add key="mlnet-daily" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/MachineLearning/nuget/v3/index.json" />
     <add key="mlnet-assets" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/machinelearning-assets/nuget/v3/index.json" />
+    <add key="nuget" value="https://api.nuget.org/v3/index.json" />
```
Comment (jeffhandley) [inline NuGet.config:16]: I'll have to defer to @ViktorHofer or @mmitche for certain, but I think we're supposed to exclusively use our dnceng feeds.

File: src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs [hunk 2]
```diff
@@ -80,14 +80,14 @@ internal unsafe string GetModelString()
             byte[] buffer = new byte[bufLen];
             int size = 0;
             fixed (byte* ptr = buffer)
-                LightGbmInterfaceUtils.Check(WrappedLightGbmInterface.BoosterSaveModelToString(Handle, 0, BestIteration, bufLen, ref size, ptr));
+                LightGbmInterfaceUtils.Check(WrappedLightGbmInterface.BoosterSaveModelToString(Handle, 0, BestIteration, 0, bufLen, ref size, ptr));
```
Comment (jeffhandley) [inline src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs:83]: Was this argument mismatch the source of the Internal CLR Error you were seeing last week?

**Review by jeffhandley — COMMENTED** [review]
I don't know how to assess the output result differences; is there anything in those changes that give you pause or concern?

File: NuGet.config (hunk 1 above)
Comment (mmitche) [reply NuGet.config:16]: Yep, this should be replaced by the "dotnet-public" feed if it's not already there.

Comment (michaelgsharp) [reply NuGet.config]: Yeah, this was just so I could get it up for review faster. The appropriate package has been mirrored and nuget is removed as a feed.

File: src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs (hunk 2 above)
Comment (michaelgsharp) [reply src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs:83]: Nope. That was a new thing added in to the new versions that we aren't using at all, but it didn't exist in the prior version. I did figure out the Internal CLR issue though just barely. :) so just waiting on builds now.

**Comment by michaelgsharp** [comment]
No concern on the output changes. Besides the fact that the training size is so small that anything (sneezes included) can change the output, these results are 100% done by the third party library, so we are ok with all changes that come from their changes.

**Review by JakeRadMSFT — APPROVED** [review]

**Pull request merged**